sdk/eventhub/event-hubs/src/eventHubClient.ts

```diff
@@ -88,6 +97,23 @@ export interface ClientOptions extends ClientOptionsBase {
   tokenProvider?: TokenProvider;
 }
 
+/**
+ * Describes the options that can be provided while creating the EventHub Client.
+ * @interface EventHubClientOptions
+ */
+export interface EventHubClientOptions<T> extends ClientOptions {
```

Comment:
`T` here is very specific to WebSocket... it feels weird to have that against EventHubClientOptions

Should we have a `WebSocketOptions<T>` instead?

Reply:
I'm not sure exactly what you're suggesting, but I believe unless the generic type is declared at the method signature (client constructor) and plumbed all the way through, inference won't happen as we want it to...

Reply:
I was using `WebSocketOptions<T>` name before but I thought, in case of non web socket scenario when user will use `createFromConnectionString<T>(connectionString: string, path?: string, options?: EventHubClientOptions<T>)` method, then this name `EventHubClientOptions` will indicate only Event Hub Client Options, not specific to web socket and all.